packages/form-js-editor/src/FormEditor.js

```diff
@@ -3,24 +3,30 @@ import Ids from 'ids';
 import { isString, set } from 'min-dash';
 
 import core from './core';
-
 import EditorActionsModule from './features/editor-actions';
 import KeyboardModule from './features/keyboard';
 import ModelingModule from './features/modeling';
 import SelectionModule from './features/selection';
+import { ALL_COMPONENTS } from './render/components/properties-panel/Util';
 
 const ids = new Ids([ 32, 36, 1 ]);
 
 /**
  * @typedef { import('./types').Injector } Injector
  * @typedef { import('./types').Module } Module
  * @typedef { import('./types').Schema } Schema
- *
+ * @typedef { import('./types').ComponentTypes} ComponentTypes
+ * @typedef { import('./types').ComponentActions} ComponentActions
  * @typedef { import('./types').FormEditorOptions } FormEditorOptions
  * @typedef { import('./types').FormEditorProperties } FormEditorProperties
  *
+ *
+ *
+ *
  * @typedef { {
  *   properties: FormEditorProperties,
+ *   availableComponentTypes?: ComponentTypes[],
```

Comment:
My proposal is to simplify this API to `allowedComponents=string[]|(name:string) => boolean`.

Reply:
Or for the initial case simply start with `allowedComponents=string[]`.